### ISSUE TITLE
Keeping subsystems decoupled and identical

### ISSUE THREAD

**Issue by NnktYoshioka** [netket/netket#305]
Hi, I am wondering how to keep the subspaces decoupled and also identical.
In other words, I want to keep two identical copy of RBM during optimization.

Initially I was thinking of using `edges` and `automorphisms` in `nk.Graph.CustomGraph`, but currently not working properly enough (using v2.1, macOS 10.14).

Concretely for 4 site system, I define the ansatz in the following way:
```python
import netket  as nk

# setup Hilbert space
edges = [(0,1), (2,3)]
automorphisms = [[0,1,2,3], [2,3,0,1]]
graph = nk.graph.CustomGraph(edges = edges, automorphisms = automorphisms)
hi = nk.hilbert.Spin(graph = graph, s=0.5, )

# rbm machine
rbm = nk.machine.RbmSpin(hi, alpha=1)
rbm.init_random_parameters(sigma=0)  
```
The option `init_random_parameters` seems to introduce coupling between all sites so it is set to zero.
When I run the optimization for some Hamiltonian, however, the two subsystems would be coupled with each other.
Is there any way to get around this?
Thanks in advance!

**Comment by PhilipVinc** [comment]
Is this to simulate Open Systems?

You could just set the interaction weights to 0 after each iteration, though this won't play well with SR.

**Comment by NnktYoshioka** [comment]
Yes, it could be used to simulate open systems, and also to consider some penalty term like <\psi|F|\psi> * <\psi | F | \psi>.
Setting interaction to zero is certainly an option. Would be even nicer if we could impose the restriction, since we could easily handle more complicated constrained optimization problems :)

**Comment by gcarleo** [comment]
Thanks @NnktYoshioka , this is a good point indeed. What we would need is a generic way to define the topology (the graph, if you want) of the networks, similar to what we do now to define the topology of the Hamiltonian. No, this is not implemented, but ideally it could be done passing a `graph` object to `machine`... I suspect that this can be already emulated using our ConvNet networks, but you would need to have partial pooling, that is not implemented as a layer... In any case, for the open system we have plans to implement the density matrix optimization etc.

**Comment by gcarleo** [comment]
Update: this should be easily implementable in v3.0
